src/main/java/htsjdk/samtools/SAMTag.java

```diff
@@ -105,24 +107,46 @@
     U2,
     UQ;
 
-    private final short shortValue = SAMTag.makeBinaryTag(this.name());;
+    // Cache of already-converted tags.  Should speed up SAM text generation.
+    // Not synchronized because race condition is not a problem.
+    private static final String[] stringTags = new String[Short.MAX_VALUE];
+
+    private final short shortValue = SAMTag.makeBinaryTag(this.name());
 
     /**
      * Convert from String representation of tag name to short representation.
      *
      * @param tag 2-character String representation of a tag name.
      * @return Tag name packed as 2 ASCII bytes in a short.
      */
-    static short makeBinaryTag(String tag) {
+    public static short makeBinaryTag(String tag) {
         if (tag.length() != 2) {
             throw new IllegalArgumentException("String tag does not have length() == 2: " + tag);
         }
         return (short)(tag.charAt(1) << 8 | tag.charAt(0));
     }
 
+    /**
+     * Convert from short representation of tag name to String representation.
+     *
+     * @param tag Tag name packed as 2 ASCII bytes in a short.
+     * @return 2-character String representation of a tag name.
+     */
+    public static String makeStringTag(final short tag) {
+        String ret = stringTags[tag];
+        if (ret == null) {
+            final byte[] stringConversionBuf = new byte[2];
+            stringConversionBuf[0] = (byte)(tag & 0xff);
+            stringConversionBuf[1] = (byte)((tag >> 8) & 0xff);
+            ret = StringUtil.bytesToString(stringConversionBuf);
+            stringTags[tag] = ret;
+        }
+        return ret;
+    }
+
     /**
      * Get the binary representation of this tag name.
-     * @see SAMTagUtil#makeBinaryTag(String)
+     * @see SAMTag#makeBinaryTag(String)
```

Comment:
A minor point, but the class prefix is unnecessary if you're linking to another member of the same class.
```suggestion
     * @see #makeBinaryTag(String)
```

Reply:
good point.